app/screens/WelcomeScreen.tsx

```diff
@@ -1,23 +1,38 @@
-import React from "react"
+import React, { useLayoutEffect } from "react"
 import { Dimensions, Image, ImageStyle, Platform, TextStyle, View, ViewStyle } from "react-native"
 import { SafeAreaView } from "react-native-safe-area-context"
 import { Button, Screen, Text } from "../components"
 import { useAppNavigation } from "../hooks"
 import { AppStackScreenProps } from "../navigators"
 import { colors, spacing } from "../theme"
+import messaging from "@react-native-firebase/messaging"
 
 const welcomeLogo = require("../../assets/images/welcome-shapes.png")
 const { width: screenWidth } = Dimensions.get("screen")
 
 interface WelcomeScreenProps extends AppStackScreenProps<"Welcome"> {}
 
+const requestUserPermission = async () => {
+  const authStatus = await messaging().requestPermission()
+  const enabled =
+    authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+    authStatus === messaging.AuthorizationStatus.PROVISIONAL
+  if (enabled) {
```

Comment:
Should we alert anything if they have denied permissions and come back to this screen?

Reply:
Honestly, in my past experience usually if a user doesn't accept one of these modals, its a burden to keep coming back to it. We might just leave it as is but also have a convo with the rest of the team if we want to go down a specific route